src/prefect/runner/submit.py

```diff
@@ -35,10 +36,30 @@ async def get_current_run_count() -> int:
         return response.json()
 
 
+async def run_prefect_callable_and_retrieve_run_id(
+    prefect_callable: Union[Flow, Task],
+    parameters: Dict[str, Any],
+) -> uuid.UUID:
+    flow_run_id: Optional[uuid.UUID] = None
+
+    def store_run_id(flow, flow_run, state):
+        nonlocal flow_run_id
+        flow_run_id = flow_run.id
+
+    prefect_callable.on_completion = (prefect_callable.on_completion or []) + [
```

Comment:
we'll probably need something better than this

Reply:
This is honestly better than I was imagining!